homeassistant/components/comfoconnect/climate.py

```diff
@@ -0,0 +1,132 @@
+"""Platform to control a Zehnder ComfoAir Q350/450/600 ventilation unit."""
+import logging
+
+from pycomfoconnect import (
+    CMD_BYPASS_OFF,
+    CMD_BYPASS_ON,
+    SENSOR_BYPASS_STATE,
+    SENSOR_TEMPERATURE_SUPPLY,
+)
+
+from homeassistant.components.climate import (
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    TEMP_CELSIUS,
+    ClimateEntity,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN, SIGNAL_COMFOCONNECT_UPDATE_RECEIVED, ComfoConnectBridge
+
+_LOGGER = logging.getLogger(__name__)
+HVAC_MODES = [HVAC_MODE_HEAT, HVAC_MODE_COOL]
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the ComfoConnect climate platform."""
+    ccb = hass.data[DOMAIN]
+
+    add_entities([ComfoConnectBypass(ccb.name, ccb)], True)
+
+
+class ComfoConnectBypass(ClimateEntity):
+    """Representation of the ComfoConnect bypass platform."""
+
+    def __init__(self, name, ccb: ComfoConnectBridge) -> None:
+        """Initialize the ComfoConnect bypass."""
+        self._name = name
+        self._ccb = ccb
+        # Write some initial values
+        self._ccb.data[SENSOR_BYPASS_STATE] = 0
+        self._ccb.data[SENSOR_TEMPERATURE_SUPPLY] = 0
```

Comment:
I think yes, without that after HA startup - it shows random values on the widget before it get fresh data from comfoconnect. Imo it's better to show 0 instead of random value.

Reply:
This is changing values in the ComfoConnectBridge, and should not be done this way.

Either, adjust the upstream library to have sane initial values, or, use a local variable that gets updated and has an initial value.

Reply:
It's been a long time since I wrote this cofe, but IIRC, a callback is made as soon as there are real values from the device. You shouldn't directly read the values from the library, and definitely shouldn't change them.

Reply:
Without this initial values I'm getting error after HA startup:
```
Jan 18 18:29:48 srv hass[1342376]: Traceback (most recent call last):
Jan 18 18:29:48 srv hass[1342376]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 316, in async_add_entities
Jan 18 18:29:48 srv hass[1342376]:     await asyncio.gather(*tasks)
Jan 18 18:29:48 srv hass[1342376]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in _async_add_entity
Jan 18 18:29:48 srv hass[1342376]:     await entity.add_to_platform_finish()
Jan 18 18:29:48 srv hass[1342376]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 531, in add_to_platform_finish
Jan 18 18:29:48 srv hass[1342376]:     self.async_write_ha_state()
Jan 18 18:29:48 srv hass[1342376]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 296, in async_write_ha_state
Jan 18 18:29:48 srv hass[1342376]:     self._async_write_ha_state()
Jan 18 18:29:48 srv hass[1342376]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 320, in _async_write_ha_state
Jan 18 18:29:48 srv hass[1342376]:     sstate = self.state
Jan 18 18:29:48 srv hass[1342376]:   File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 173, in state
Jan 18 18:29:48 srv hass[1342376]:     return self.hvac_mode
Jan 18 18:29:48 srv hass[1342376]:   File "/usr/src/homeassistant/homeassistant/components/comfoconnect/climate.py", line 110, in hvac_mode
Jan 18 18:29:48 srv hass[1342376]:     bypass = self._ccb.data[SENSOR_BYPASS_STATE]
```

It comes from this function:
```
@property
    def hvac_mode(self):
        """Return the current bypass mode. If the bypass state > 0 ComfoConenct started enabling bypass. 100% of bypass state = minimal heat recovery."""
        bypass = self._ccb.data[SENSOR_BYPASS_STATE]
        if bypass:
            _LOGGER.debug(
                "Current bypass state: %s, return: %s", bypass, HVAC_MODE_COOL
            )
            return HVAC_MODE_COOL

        _LOGGER.debug("Current bypass state: %s, return: %s", bypass, HVAC_MODE_HEAT)
        return HVAC_MODE_HEAT
```

Reply:
You shouldn't access the _ccb.data directly, but save it when you get the callback.

Reply:
Take a look async_added_to_hass and the dispatcher code.

Reply:
Updated with last commit